Updated stylings to work with mobile

kusalekanayake opened the pull request:
I've updated the site's mobile experience thanks to the ability to be able to use variants and selectors within tailwind and the css of the file. I haven't added any new content, but just made the current content more manageable. I've updated the stencila theme on the main repo as well for a few issues.

![image](https://github.com/user-attachments/assets/50e6f1a0-e870-46ff-a9bb-08931c28de8c)

kusalekanayake committed:
Updated stylings to work with mobile

Padding and size is more mobile friendly. Needs stencila theme on main repo to also be updated.

kusalekanayake commented:
<img width="2672" alt="image" src="https://github.com/user-attachments/assets/a00075cb-037c-4d89-86ba-e4891b2cf783">
It is possible to add the background image now too.

mike-parkin commented on main.smd:
`````diff
@@ -5,9 +5,15 @@ description: Documents with scientific intelligence
 title: Stencila
 ---
 
+```` css raw
+ 
+
+
+````
+
`````
This empty block is showing up at the top, just behind the header when you expand the cards, just wandering if its required for a hack of some kind, or can it be removed

mike-parkin commented on main.smd:
`````diff
@@ -18,7 +24,7 @@ title: Stencila
 ````html raw
-<div class="flex items-center justify-end gap-x-4">
+<div class="flex items-center justify-end gap-x-4 pr-12">
   <a href="https://github.com/stencila/stencila#readme" class="header-link"><img  src="images/github.svg" alt="github" class="w-6"/></a>
   <a href="https://discord.gg/GADr6Jv" class="header-link"><img  src="images/discord.svg" alt="discord" class="w-6"/></a>
 </div>
`````
I think these 'header-link' classes can be removed? unless they're used in the stencila theme stylesheet.

mike-parkin commented on main.smd:
`````diff
@@ -132,63 +142,69 @@ In the fast-paced world of AI, keeping up with the latest and greatest models ca
 
 ````html raw
-<div class="flex justify-center space-x-48 text-center">
+<div class="block md:flex justify-center text-center max-auto w-full space-y-16">
`````
the 'max-auto' utility class, is this meant to be mx-auto?

mike-parkin reviewed (commented):
I think it looks way better than before, and responds really well.

I may not be using the right theme css (using the lastest origin main branch). so some of these might be redundant.

The footer is a little crowded on sm devices.

![image](https://github.com/user-attachments/assets/6bcc3155-2a96-4c48-afaa-fd13a42ecb8a)

And could possibly use a keyline under the header when scrolling.

![image](https://github.com/user-attachments/assets/6d3481d3-8e65-4375-bf53-fc90c98269e0)

kusalekanayake committed:
Tidied up some extra classes

PR comments

kusalekanayake committed:
Added the slightest of keylines

For the header

kusalekanayake commented:
Thanks @mike-parkin! I've updated the PR accordingly. I'm not super happy with the footer just yet, I've removed the plans section for now on mobile. This may be about as far as i can take it without spending too much more time on it.
<img width="408" alt="image" src="https://github.com/user-attachments/assets/c0506275-6019-4230-a19f-7a8b7b0496eb">
<img width="411" alt="image" src="https://github.com/user-attachments/assets/a5fa254e-e083-4da2-8eb4-e96ab6f786d8">

kusalekanayake committed:
Added headers for web preview
